From 939b129e24c0ad5d51368b1aa72fffcaca76f06f Mon Sep 17 00:00:00 2001
From: alhudz <al.hudz.k@gmail.com>
Date: Mon, 1 Jun 2026 19:05:57 +0530
Subject: [PATCH] set group value from child tokens to avoid quadratic grouping

---
 sqlparse/sql.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sqlparse/sql.py b/sqlparse/sql.py
index 6b21859..5489f05 100644
--- a/sqlparse/sql.py
+++ b/sqlparse/sql.py
@@ -164,7 +164,7 @@ class TokenList(Token):
     def __init__(self, tokens=None):
         self.tokens = tokens or []
         [setattr(token, 'parent', self) for token in self.tokens]
-        super().__init__(None, str(self))
+        super().__init__(None, ''.join(token.value for token in self.tokens))
         self.is_group = True
 
     def __str__(self):
@@ -327,7 +327,7 @@ class TokenList(Token):
             grp = start
             grp.tokens.extend(subtokens)
             del self.tokens[start_idx + 1:end_idx]
-            grp.value = str(start)
+            grp.value += ''.join(token.value for token in subtokens)
         else:
             subtokens = self.tokens[start_idx:end_idx]
             grp = grp_cls(subtokens)
-- 
2.50.1 (Apple Git-155)

